### PR TITLE
NEPT-1071: Implement the new header search bars component Search form component.

### PR DESCRIPTION
**Pull request by gillesdeudon**

### CODE REVIEW

**Comment by imanoleguskiza** [comment]
👍

File: templates/button/button.component.inc [hunk 1]
```diff
@@ -134,8 +134,12 @@ function europa_atomium_definition_button() {
  * Implements hook_preprocess_HOOK().
  */
 function europa_preprocess_button(&$variables) {
-  $variables['attributes_array']['class'][] = 'ecl-button';
-  if (empty(array_search('ecl-button--', $variables['attributes_array']['class']))) {
+  if (empty($variables['attributes_array']['class'])) {
```
Comment (voidtek) [inline templates/button/button.component.inc]: This modification should be in the ticket NEPT-1060

File: template.php [hunk 2]
```diff
@@ -100,5 +100,27 @@ function _europa_form_set_css_class(array &$element, array $classes = array()) {
...
+ * @param array $haystack
+ *   The array to search in.
+ *
+ * @return mixed
+ *   The key for needle if it is found in the
+ *   array, FALSE otherwise.
+ *
+ * @author Bran van der Meer <branmovic@gmail.com>
+ */
+function _europa_array_find($needle, array $haystack) {
```
Comment (gillesdeudon) [inline template.php]: Nope, it is not the same.

array_search does not allow to make searches on partial string while this one yes.

File: templates/button/button.component.inc (hunk 1 above)
Comment (gillesdeudon) [inline templates/button/button.component.inc]: And block my ticket, no way.

**Comment by Fefaine** [comment]
👍

**Pull request merged**